docs/metrics.md

```diff
@@ -0,0 +1,120 @@
+# Metrics
+
+Test metrics are sent to 3 different places:
+1. reported properies
+2. test telemetry message bodies
+3. Azure Monitor
+
+Not all metrics are sent to all destinations.
+
+| group | reported properties | test telemetry | Azure Monitor |
+| Session Metrics | X | X | - |
+| Test Metrics | X | X | X |
+| System Health Metrics | - | X | X |
+
+## Session Metrics
+
+Some session metrics are recorded at the beginning of the run and others are recorded throughout the run.
+They record details on the progress of the current run.
+
+```json
+  {
+    "reported": {
+      "thief": {
+        "sessionMetrics": {
+          "exitReason": "Main thread raised <class KeyboardInterrupt>",
+          "latestUpdateTimeUtc": "2020-12-14T21:14:57.447061",
+          "pairingId": "176e7469-9755-4a7a-af74-85f735f0f296",
+          "runEndUtc": "2020-12-14T21:14:57.441870+00:00",
+          "runStartUtc": "2020-12-14T21:13:22.430280+00:00",
+          "runState": "interrupted",
+          "runTime": "0:01:35.016760"
+        }
+      }
+    }
+  }
+```
+
+| field | format | meaning |
+| - | - | - |
+| `runState` | string | State of the current run.  One of `waiting`, `running`, `failed`, `complete`, or `interrupted` |
+| `runStartUtc` | string | DateTime in UTC for the start of the current run. |
+| `runEndUtc` | string | DateTime in UTC for the end of the current run.  Only valid for `failed`, `complete`, or `interrupted`. |
+| `runTime` | string | Elapsed time for current run |
+| `latestUpdateTimeUtc` | string | DateTime in UTC for most recent update to thie structure. |
+| `exitReason` | string | freeform string indicating reason for test to exit.  Most likely exception text or other error string.  Only valid for `failed`, `complete`, or `interrupted`. |
+| `pairingId` | guid | `pairingId` for the current run |
+
+## Test Metrics
+
+Test metrics are recorded throughout the run and they record statistics on the various test operations that are occuring.
+Most or all of the test metrics overlap with metrics that are sent in telemetry and pushed to Azure Monitor.
+
+```json
+  {
+    "reported": {
+      "thief": {
+        "testMetrics": {
+          "receiveMessageCountMissing": 0,
```

Comment:
For Azure monitor standards, I'd go with TitleCase names here

Reply:
But camelCase is standard for json, which this is.  You can't win.  Is it worth doing camelCase inside json and TitleCase for Azure Monitor or should we just pick one and stick with it.  (I'm already dealing with camelCase and boxcar_case to honor python standards).

Reply:
Ah, I understand. When we send it as telemetry, it should be camel-cased, yes.

When we send it to Azure Monitor, we can consider if we want TitleCase or CamelCase.

Reply:
will change in later PR